lxd/metrics/api_rates.go

```diff
@@ -0,0 +1,93 @@
+package metrics
+
+import (
+	"net/url"
+	"sync/atomic"
+
+	"github.com/canonical/lxd/shared/entity"
+)
+
+// RequestStatus represents the a completed request status category
+type requestStatus string
+
+const (
+	serverError requestStatus = "server_error"
+	clientError requestStatus = "client_error"
+	success     requestStatus = "succeeded"
+)
+
+// PossibleRequestStatus is a slice that includes all the possible request status categories
+var PossibleRequestStatus = []requestStatus{
+	serverError,
+	clientError,
+	success,
+}
+
+func resolveStatusCode(statusCode int, asyncOperation bool) requestStatus {
+	if asyncOperation {
+		if statusCode == 400 {
+			return serverError
+		}
+		return success
+	}
+	if statusCode > 500 && statusCode < 600 {
+		return serverError
+	}
+	if statusCode > 400 && statusCode < 500 {
+		return clientError
+	}
+	return success
+}
+
+func resolveEntityType(url url.URL) entity.Type {
+	entityType, _, _, _, _ := entity.ParseURL(url)
+	if entityType == entity.TypeContainer { // Handle containers as TypeInstance
+		return entity.TypeInstance
+	}
+	return entityType
+}
+
+type completedMetricsLabeling struct {
+	entityType entity.Type
+	status     requestStatus
+}
+
+var ongoingRequests map[entity.Type]*int64
```

Comment:
why pointer to int64?

Reply:
This is for performing atomic operations on these values later on the code, I believe pointing to values on a map is not possible due to how Go handles maps internally.

Reply:
True, but the maps are passed by reference so you can just refer to the map directly.

Reply:
Sorry, I am failing to see how that helps accessing the address for the map values. Could you please rephrase to help me understand?